nodebuilder/state/keyring.go

```diff
@@ -16,15 +16,19 @@ type AccountName string
 func Keyring(cfg Config, ks keystore.Keystore) (kr.Keyring, AccountName, error) {
 	ring := ks.Keyring()
 	var info *kr.Record
-	// if custom keyringAccName provided, find key for that name
-	if cfg.KeyringAccName != "" {
-		keyInfo, err := ring.Key(cfg.KeyringAccName)
+
+	for _, accName := range cfg.KeyringAccName {
+		keyInfo, err := ring.Key(accName)
 		if err != nil {
 			log.Errorw("failed to find key by given name", "keyring.accname", cfg.KeyringAccName)
 			return nil, "", err
 		}
-		info = keyInfo
-	} else {
+		if info == nil {
+			info = keyInfo
+		}
+	}
+
+	if info == nil {
```

Comment:
It looks like keyInfo variable gets overwritten on on every iteration over KeyringAccName. So only the last one will be used. What is the reason to have multiple accounts specified, if only one will be used? If there is some logic implied, lets add comment for it.

Reply:
it does not. info is set up on the first iteration of the loop or the default account is used in case loop is empty.

Reply:
@vgonkivs default key should only be used in the case that no keys are specified (len(cfg.KeyringAccName) == 0)

Reply:
also please document what is happening here (just checking that keys that are specified actually exist in keyring)

Reply:
> @vgonkivs default key should only be used in the case that no keys are specified (len(cfg.KeyringAccName) == 0)

if (len(cfg.KeyringAccName) == 0), then info == nil.